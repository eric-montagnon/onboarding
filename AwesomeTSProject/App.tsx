/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from 'react';
import {StyleSheet, Text, View} from 'react-native';

const App = () => {
  return (
    <View style={styles.cont}>
      <View style={styles.boxstyle}>
        <Text style={styles.text}> Hello !</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  cont: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  boxstyle: {
    justifyContent: 'center',
    alignItems: 'center',
    height: 200,
    padding: 20,
    width: 200,
    backgroundColor: 'red',
    borderRadius: 100 / 2,
  },
  text: {
    color: 'green',
    alignItems: 'center',
    fontWeight: 'bold',
    fontSize: 25,
  },
});

export default App;
