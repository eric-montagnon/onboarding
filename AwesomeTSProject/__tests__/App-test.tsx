/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import '@testing-library/jest-native/extend-expect';
import {render, screen, waitFor} from '@testing-library/react-native';

it('renders correctly', async () => {
  const nock = require('nock');
  nock('https://www.thecocktaildb.com')
    .get('/api/json/v1/1/filter.php')
    .reply(200, {
      drink: [
        {
          idDrink: 12,
          strDrink: 'OKAY',
          strDrinkThumb:
            'https://www.thecocktaildb.com/images/media/drink/rrtssw1472668972.jpg',
          strCategory: 'Classic drink',
        },
      ],
    });
  render(<App />);

  await waitFor(() => {
    expect(screen.queryByTestId('loader')).not.toBeOnTheScreen();
  });

  expect(screen).toMatchSnapshot();
});
