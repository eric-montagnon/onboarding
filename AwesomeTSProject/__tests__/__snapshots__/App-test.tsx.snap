// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 50,
        "height": 200,
        "justifyContent": "center",
        "padding": 20,
        "width": 200,
      }
    }
  >
    <Text
      style={
        {
          "alignItems": "center",
          "color": "green",
          "fontSize": 25,
          "fontWeight": "bold",
        }
      }
    >
       Hllo !
    </Text>
  </View>
</View>
`;
